### PR TITLE
Adding Trata to conda-forge

### PR DESCRIPTION
**Pull request by doutriaux1**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by olson59** [comment]
I'm willing to be listed as a maintainer.

**Comment by doutriaux1** [comment]
@conda-forge-admin, please lint

**Comment by doutriaux1** [comment]
@conda-forge/help-python, ready for review!

File: recipes/trata/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (ocefpaf) [inline recipes/trata/meta.yaml:1]: Ping!

File: recipes/trata/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  # It is okay if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
+  # noarch: python
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  # More info about selectors can be found in the conda-build docs: 
+  # https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (ocefpaf) [inline recipes/trata/meta.yaml:30]: This one can probably be noarch.

```suggestion
    noarch: python
    number: 0
```

File: recipes/trata/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+# If your package is python based, we recommend using Grayskull to generate it instead:
+# https://github.com/conda-incubator/grayskull
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 16 is convenient
+# when copying and pasting from another recipe, but not really needed.
+{% set name = "trata" %}
+{% set version = "1.0.2" %}
```
Comment (ocefpaf) [inline recipes/trata/meta.yaml:9]: It is tagged as 1.0.2 but the metadata in 1.0.1. See https://github.com/LLNL/trata/blob/95b9a404d0382f233c7e0a4e73f8b528b2bd1f3f/pyproject.toml#L4

File: recipes/trata/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  doc_url: https://trata.readthedocs.io/
+  dev_url: https://github.com/trata/trata
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
```
Comment (ocefpaf) [inline recipes/trata/meta.yaml:50]: You can remove this comment too.

**Comment by doutriaux1** [comment]
@ocefpaf little ping here. I think we're good

File: recipes/trata/meta.yaml (hunk 3 above)
Comment (ocefpaf) [reply recipes/trata/meta.yaml:2]: I'm not sure what is the correct version 1.0.1 or 1.0.2? The metadata is different from the one here.

File: recipes/trata/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/trata/meta.yaml:14]: It does look like noarch. Can you please confirm?

**Comment by doutriaux1** [comment]
@ocefpaf another ping.

**Comment by xhochy** [comment]
There are still open comments from @ocefpaf. Please address them.

File: recipes/trata/meta.yaml (hunk 2 above)
Comment (doutriaux1) [reply recipes/trata/meta.yaml:14]: no arch didn't work

File: recipes/trata/meta.yaml (hunk 3 above)
Comment (doutriaux1) [reply recipes/trata/meta.yaml:2]: 1.0.2 this has been fixed now

**Comment by ocefpaf** [comment]
> @ocefpaf another ping.

@doutriaux1 please take a look at the comments above. They are still unresolved.

**Comment by doutriaux1** [comment]
@ocefpaf @xhochy

**Comment by doutriaux1** [comment]
> > @ocefpaf another ping.
> 
> @doutriaux1 please take a look at the comments above. They are still unresolved.

I believe they are all resolved now.

File: recipes/trata/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/trata/meta.yaml:14]: What failed?

File: recipes/trata/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  host:
+    - python
+    - pip
+    - poetry
+  run:
+    - python
+    - scikit-learn
+    - numpy
+    - scipy
+    - matplotlib
```
Comment (ocefpaf) [inline recipes/trata/meta.yaml:26]:
```suggestion
    - matplotlib-base
```

File: recipes/trata/meta.yaml (hunk 2 above)
Comment (doutriaux1) [reply recipes/trata/meta.yaml:14]: @ocefpaf please see: https://github.com/conda-forge/staged-recipes/runs/21926254364

Comment (ocefpaf) [reply recipes/trata/meta.yaml:14]: We can't. Azure doesn't keep the logs long enough. Can you please describe it or try it again?

Screenshot from 2024-03-27 15-11-27


PS: It is OK for noarch package to fails on Windows with a missing dependency. We prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows.

**Comment by doutriaux1** [comment]
@ocefpaf you can now see the windows error, I am not sure what the problem is....

**Comment by ocefpaf** [comment]
> @ocefpaf you can now see the windows error, I am not sure what the problem is....

It is in the logs:

```
Encountered problems while solving:
  - nothing provides requested pytorch
  - nothing provides requested torchvision
```

Like I mentioned above, it is fine for noarch packages to fail b/c of a missing dependency. We prefer noarch to save resources, and when `pytorch` and `torchvision` are available on Windows, you don't need to do anything here b/c this package will be ready for Windows.

**Comment by doutriaux1** [comment]
@ocefpaf ok. I'm guessing they come from scikit-learn then.

**Comment by ocefpaf** [comment]
> @ocefpaf ok. I'm guessing they come from scikit-learn then.

It should not but I overlooked that there are many other PRs accumulated here (I just got back from travel and I;m reviewing tons os PRs). So we may need to wait for staged-recipes to clear up and update the branch here to see the log for trata only.

**Comment by ocefpaf** [comment]
@doutriaux1 apologies but my kid clicked here b/c I left my laptop open :-/

I'll re-do this PR for you.

**Pull request merged**